src/librustc_yk_sections/mir_cfg.rs

```diff
@@ -18,20 +16,95 @@ use rustc::ty::TyCtxt;
 
 use rustc::hir::def_id::DefId;
 use rustc::mir::{
-    Mir, TerminatorKind, Operand, Constant, StatementKind, BasicBlock, BasicBlockData, Terminator
+    Mir, TerminatorKind, Operand, Constant, StatementKind, BasicBlock, BasicBlockData, Terminator,
+    Place, Rvalue, Statement, Local, PlaceBase
 };
 use rustc::ty::{TyS, TyKind, Const, LazyConst};
 use rustc::util::nodemap::DefIdSet;
 use std::path::PathBuf;
 use std::fs::File;
+use std::convert::TryFrom;
 use rustc_yk_link::YkExtraLinkObject;
 use std::fs;
 use std::error::Error;
+use std::cell::{Cell, RefCell};
+use rustc_data_structures::indexed_vec::{IndexVec, Idx};
+use rustc_data_structures::graph::dominators::Dominators;
 use ykpack;
 
 const SECTION_NAME: &'static str = ".yk_tir";
 const TMP_EXT: &'static str = ".yk_tir.tmp";
 
+type TirVarIndex = u32;
+
+/// A conversion context holds the state needed to perform the conversion to (pre-SSA) TIR.
+struct ConvCx<'a, 'tcx, 'gcx> {
+    /// The compiler's god struct. Needed for queries etc.
+    tcx: &'a TyCtxt<'a, 'tcx, 'gcx>,
+    /// The definition sites of TIR variable in terms of basic blocks.
+    def_sites: RefCell<Vec<Vec<BasicBlock>>>,
+    /// The next TIR variable index to be issued.
```

Comment:
Huh? Is this a fancy-pants yet vague way of saying "This is a monotonically increasing integer which is used to give Tir nodes a unique ID"?

Reply:
Yes, but it's current value is the one which will be given out next.

Reply:
[I'm likely to abstract this mechanism in the next PR, as the last stage requires us to re-number all of our variables in a similar fashion. The reason there is a mapping in the conversion is because I don't think we will get far with a one-to-one mapping from MIR to TIR variables, unless we also model projections in the same way as Rust does.]

Reply:
It doesn't matter if this is the current or next index: it's the monotonically increasing bit which is important.